### ISSUE TITLE
Deadlock during migration

### ISSUE THREAD

**Issue by cosmer-work** [groue/GRDB.swift#741]
### What did you do?

Updated to GRDB from 4.9.0 to 4.12.0.

### What did you expect to happen?

No deadlock.

### What happened instead?

A deadlock occurred during migration.

### Environment

**GRDB flavor(s):** GRDB
**GRDB version:** 4.12.0
**Installation method:** SPM
**Xcode version:** 11.4
**Swift version:** 5.2
**Platform(s) running GRDB:** iOS 13.4
**macOS version running Xcode:** 10.15.4

### Steps to Reproduce

Assign a target queue to the configuration
```
var configuration = Configuration()
configuration.targetQueue = targetQueue
```

Enable erase database on schema change
```
var migrator = DatabaseMigrator()
migrator.eraseDatabaseOnSchemaChange = true
```

A deadlock will occur at line 51 of SerializedDatabase.swift after calling `migrator.migrate`.
`try queue.sync {`

4.10.0 doesn't have this issue. 4.11.0 does.

**Comment by groue** [comment]
Hello @cosmer-work, thank you for the report!

**Comment by groue** [comment]
A fix has been pushed in v4.12.1

**Comment by cosmer-work** [comment]
Thank you @groue.
